e2e/run/setups/wait.go

```diff
@@ -29,25 +36,19 @@ func WaitForServerReady(baseURL string) error {
 		return err
 	}
 
-	const backoff = 200 * time.Millisecond
-	const maxWait = 5 * time.Second
-	var (
-		resp *http.Response
-		rerr error
-	)
-	for i := 0; i < int(maxWait/backoff); i++ {
-		resp, rerr = client.Do(req.Clone(context.Background()))
+	deadline := time.Now().Add(MaxWait)
+	for {
+		resp, err := client.Do(req.Clone(context.Background()))
 
 		if resp != nil && resp.StatusCode == http.StatusOK {
 			resp.Body.Close() //noline:errcheck // we don't care about the body
 			return nil
 		}
```

Comment:
BTW Aren't we leaking response body in here? We should close it regardless the status code. Could you double check?